### PR TITLE
fix statechannel consumer detection

### PR DESCRIPTION
**Pull request by ianhe8x**
* fix consumer detection when it is created by consumerhost contract
  * use try-catch, when decode thrown error, it means callback is not from consumerhost
* change terminatedAt to optional, and leave it undefined when creating the channel

### CODE REVIEW

File: src/mappings/stateChannel.ts [hunk 1]
```diff
@@ -24,30 +24,32 @@ export async function handleChannelOpen(
...
     callback,
   } = event.args;
-
-  const abi = ethers.utils.defaultAbiCoder;
-  const consumer = abi.decode(['address'], callback)[0] as string;
-  const agent = creator === consumer ? consumer : undefined;
+  let consumer = _consumer;
+  let agent: string | undefined = undefined;
+  try {
+    consumer = utils.defaultAbiCoder.decode(['address'], callback)[0] as string;
```
Comment (mzxyz) [inline src/mappings/stateChannel.ts:37]: The consumer is the `creator` of the channel in the event, the `address` decode from the callback is the real consumer (user), in that case, `agent` should equal `agent = creator !== consumer ? creator : undefined`.

Comment (ianhe8x) [reply src/mappings/stateChannel.ts:37]: if the channel is not created by contract, you won't be able to decode address from callback, it is '0x00..'. In that case, we need to use try-catch and assign consumer = creater in the catch block.

Comment (ianhe8x) [reply src/mappings/stateChannel.ts:37]: I do find an issue with my code, have fixed.

Comment (mzxyz) [reply src/mappings/stateChannel.ts:37]: Yes, that's correct, so the changes will look like this:
```
  let consumer = undefined;
  let agent: string | undefined = undefined;
  try {
    consumer = utils.defaultAbiCoder.decode(['address'], callback)[0] as string;
    agent = _consumer !== consumer ? _consumer : undefined.
  catch {
    consumer = _consumer;
  }
```

Comment (mzxyz) [reply src/mappings/stateChannel.ts:37]: @nz-neooo could you confirm that whether the `address` will always includes in the `callback` or only for hosting plan.

Comment (nz-neooo) [reply src/mappings/stateChannel.ts:37]: @mzxyz callback is work for all `ConsumerHost` created channel.

**Review by nz-neooo — APPROVED** [review]

**Review by mzxyz — APPROVED** [review]

**Pull request merged**